basics/basic-types.md

```diff
@@ -1,91 +1,97 @@
-# Basic types
+# Kiểu số cơ bản
 
-D provides a number of basic types which always have the same
-size **regardless** of the platform - the only exception
-is the `real` type which provides the highest possible floating point
-precision. There is no difference
-between the size of an integer regardless of whether the application
-is compiled for 32-bit or 64-bit systems.
+D có các kiểu số cơ bản với kích cỡ bộ nhớ cần thiết là như nhau bất kể
+hệ điều hành đang chạy là gì, ngoại trừ kiểu số thực `real` với dấu chấm động.
+Cụ thể hơn, dù bạn đang chạy ứng dụng D trên máy 32 hay 64 bit, thì
+mỗi biến của các kiểu sau vẫn chiếm cùng số bít trên bộ nhớ.
 
-| type                          | size
+| kiểu                          | cỡ
 |-------------------------------|------------
 |`bool`                         | 8-bit
 |`byte`, `ubyte`, `char`        | 8-bit
 |`short`, `ushort`, `wchar`     | 16-bit
 |`int`, `uint`, `dchar`         | 32-bit
 |`long`, `ulong`                | 64-bit
 
-#### Floating point types:
+#### Kiểu số thực với dấu chấm động
 
-| type    | size
+| kiểu    | cỡ
 |---------|--------------------------------------------------
 |`float`  | 32-bit
 |`double` | 64-bit
-|`real`   | >= 64-bit (generally 64-bit, but 80-bit on Intel x86 32-bit)
+|`real`   | >= 64-bit (thường là 64-bit, hoặc 80-bit trên Intel x86 32-bit)
 
-The prefix `u` denotes *unsigned* types. `char` translates to
-UTF-8 characters, `wchar` is used in UTF-16 strings and `dchar`
-in UTF-32 strings.
+Kiểu bắt đầu bởi `u` để kiểu không nhận giá trị âm (không dấu).
+Kiểu `char` thực sự được nhúng trong UTF-8, trong khi `wchar`
+dùng ký tự UTF-16, còn `dchar` là UTF-32.
 
-A conversion between variables of different types is only
-allowed by the compiler if no precision is lost. However,
-a conversion between floating point types
-(e.g `double` to `float`) is allowed.
+Trình biên dịch D chỉ thực hiện chuyển đổi kiểu nếu đảm bảo được sự toàn vẹn
```

Comment:
I think `cho phép` has a more precise meaning than `thực hiện` in this case.

Reply:
Fixed in translation.